src/hotspot/share/oops/instanceKlass.cpp

```diff
@@ -528,7 +528,7 @@ InstanceKlass::InstanceKlass(const ClassFileParser& parser, KlassKind kind, Refe
 }
 
 void InstanceKlass::deallocate_methods(ClassLoaderData* loader_data,
-                                       Array<Method*>* methods) {
+                                       Array<Method*>* methods, InstanceKlass* klass) {
```

Comment:
An explicit boolean parameter would be cleaner/clearer.

Reply:
You mean something like 
```
void InstanceKlass::deallocate_methods(ClassLoaderData* loader_data,
                                       Array<Method*>* methods) {
                                       Array<Method*>* methods, InstanceKlass* klass, bool has_class) {
 ```
 ?

Reply:
I just removed the `klass` argument. It is not really used anyway.

Reply:
I actually ended up with a boolean parameter here.